src/pages/Environments.js

```diff
@@ -456,8 +456,10 @@ const Environments = () => {
   // created, workspace namespace (a.k.a billing project) value used to equal the google project.
   // Therefore we use google project if the namespace label is not defined.
   const renderWorkspaceForApps = app => {
-    const { appType, googleProject, labels: { saturnWorkspaceNamespace = googleProject, saturnWorkspaceName } } = app
-    return getWorkspaceCell(saturnWorkspaceNamespace, saturnWorkspaceName, appType, false)
+    const { appType, googleProject, cloudContext: { cloudResource }, labels: { saturnWorkspaceNamespace = googleProject, saturnWorkspaceName } } = app
+    // Here, we use the saturnWorkspaceNamespace label if its defined, otherwise the googleProject if its defined, otherwise the cloudResource
+    const resolvedSaturnWorkspaceName = saturnWorkspaceNamespace ? saturnWorkspaceNamespace : cloudResource
```

Comment:
Should this variable be called `resolvedSaturnWorkspaceNamespace` instead? namespace =/= name, right?

Reply:
nit: it might also make these lines a little bit clearer to remove the ` = googleProject` default value from the destructured assignment on line 459, and instead make this line `const resolvedThing = saturnWorkspaceNamespace || googleProject || cloudResource`, to make the triple fallback obvious